vspace/src/main/java/edu/asu/diging/vspace/core/data/ModuleRepository.java

```diff
@@ -24,4 +29,10 @@ Page<IModule> findDistinctByNameContainingOrDescriptionContaining(Pageable reque
     List<Module> findAllByOrderByCreationDateDesc();
     
     List<Module> findAllByModuleStatus(ModuleStatus moduleStatus);
+    
+    List<ISequence> findAllByStartSequenceNotNull();
+       
+    @Query("SELECT DISTINCT module from Module module JOIN ModuleLink moduleLink ON module.id = moduleLink.module.id JOIN Space space on space.id = moduleLink.space.id where module.moduleStatus =?1 AND space.spaceStatus = ?2 AND ( module.name like %?3% OR module.description like %?4% )") 
+    Page<IModule> findDistinctByModuleStatusNameContainingOrDescriptionContainingLinkedToSpace(Pageable requestedPage, 
```

Comment:
If there is a query annotation, the method name really didn't need to be that long. I'm sure there is a shorter name that conveys what the method is doing that's not that long.